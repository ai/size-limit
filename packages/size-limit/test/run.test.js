let { join } = require('path')

let run = require('../run')

jest.mock('../../time/get-running-time', () => () => 1)

jest.mock('../../time/cache', () => ({
  getCache () {
    return false
  },
  saveCache () {}
}))

jest.mock('ora', () => {
  return () => ({
    start () {
      return this
    },
    succeed () {},
    fail () {}
  })
})

const TMP_DIR = /size-limit-[\w-]+\/?/g
const ROOT = join(__dirname, '..', '..', '..')

function fixture (...files) {
  return join(__dirname, 'fixtures', ...files)
}

function createProcess (cwd, args = []) {
  let history = {
    exitCode: 0,
    stdout: '',
    stderr: ''
  }
  let process = {
    argv: ['node', 'size-limit', ...args],
    cwd () {
      if (cwd.includes('/')) {
        return cwd
      } else {
        return fixture(cwd)
      }
    },
    exit (code) {
      history.exitCode = code
    },
    stdout: {
      write (str) {
        history.stdout += str
          .split(ROOT)
          .join('')
          .replace(TMP_DIR, '')
      }
    },
    stderr: {
      write (str) {
        history.stderr += str
          .split(ROOT)
          .join('')
          .replace(TMP_DIR, '')
      }
    }
  }
  return [process, history]
}

function clean (output) {
  return output
    .replace(/var\/folders\/(.*)\//g, 'tmp/')
    .replace(/"cwd": "[^"]+"/, '"cwd": "/tmp/"')
    .replace(/"webpackOutput": "[^"]+"/, '"webpackOutput": "/tmp/"')
}

async function check (cwd, args) {
  let [process, history] = createProcess(cwd, args)
  await run(process)
  expect(history.stderr).toEqual('')
  expect(history.exitCode).toEqual(0)
  return history.stdout
}

async function error (cwd, args) {
  let [process, history] = createProcess(cwd, args)
  await run(process)
  expect(history.stdout).toEqual('')
  expect(history.exitCode).toEqual(1)
  return history.stderr
}

describe(`run`, () => {
  beforeEach(() => {
    jest.useFakeTimers()
  })

  it('shows version', async () => {
    let out = await check('file', ['--version'])
    expect(out).toMatch(/^size-limit \d+.\d+.\d+\n$/m)
  })

  it('shows help', async () => {
    expect(await check('file', ['--help'])).toMatchSnapshot()
  })

  it('shows webpack-related help', async () => {
    expect(await check('webpack', ['--help'])).toMatchSnapshot()
  })

  it('uses dependencies to detect plugins', async () => {
    expect(await check('dependencies', ['--help'])).toMatchSnapshot()
  })

  it('shows error on missed package.json', async () => {
    expect(await error('/')).toMatchSnapshot()
  })

  it('shows syntax errors in package.json', async () => {
    let stderr = await error('package-syntax')
    expect(stderr).toContain('ERROR')
    expect(stderr).toContain('JSONError')
  })

  it('shows error in JSON format', async () => {
    let [process, history] = createProcess('/', ['--json'])
    await run(process)
    expect(history.exitCode).toEqual(1)
    expect(history.stderr).toEqual('')
    let output = JSON.parse(history.stdout)
    expect(Object.keys(output)).toEqual(['error'])
    expect(output.error).toContain('SizeLimitError: Size Limit didnâ€™t find')
  })

  it('shows migration guide for npm users', async () => {
    expect(await error('legacy-npm')).toMatchSnapshot()
  })

  it('shows migration guide for yarn users', async () => {
    expect(await error('legacy-yarn')).toMatchSnapshot()
  })

  it('shows migration guide for npm users: config 1, dep 0', async () => {
    let [process, history] = createProcess('npm-with-config-without-dev')
    await run(process)
    expect(history.exitCode).toEqual(1)
    expect(history.stderr).toMatchSnapshot()
  })

  it('shows size-limit dependency warning', async () => {
    let [process, history] = createProcess('warn')
    await run(process)
    expect(history.exitCode).toEqual(0)
    expect(history.stderr).toMatchSnapshot()
  })

  it('throws on unknown argument', async () => {
    expect(await error('file', ['--unknown'])).toMatchSnapshot()
  })

  it('throws on --save-bundle argument without webpack', async () => {
    expect(await error('file', ['--save-bundle'])).toMatchSnapshot()
  })

  it('throws on --save-bundle argument without DIR parameter', async () => {
    expect(await error('webpack', ['--save-bundle'])).toMatchSnapshot()
    expect(
      await error('webpack', ['--save-bundle', '--clean-dir'])
    ).toMatchSnapshot()
  })

  it('throws on --clean-dir argument without --save-bundle', async () => {
    expect(await error('file', ['--clean-dir'])).toMatchSnapshot()
  })

  it('throws on --why argument without webpack', async () => {
    expect(await error('file', ['--why'])).toMatchSnapshot()
  })

  it('throws on no config', async () => {
    expect(await error('file')).toMatchSnapshot()
  })

  it('throws on non-array config', async () => {
    expect(await error('non-array')).toMatchSnapshot()
  })

  it('throws on empty config', async () => {
    expect(await error('empty')).toMatchSnapshot()
  })

  it('throws on non-object check', async () => {
    expect(await error('non-object')).toMatchSnapshot()
  })

  it('throws on non-string path', async () => {
    expect(await error('non-string')).toMatchSnapshot()
  })

  it('throws on non-string entry', async () => {
    expect(await error('non-string-entry')).toMatchSnapshot()
  })

  it('throws on webpack option without webpack plugin', async () => {
    expect(await error('non-webpack')).toMatchSnapshot()
    expect(await error('non-webpack-ignore')).toMatchSnapshot()
    expect(await error('non-webpack-config')).toMatchSnapshot()
  })

  it('throws on gzip option without gzip plugin', async () => {
    expect(await error('non-gzip')).toMatchSnapshot()
  })

  it('throws on running option without time plugin', async () => {
    expect(await error('non-time')).toMatchSnapshot()
  })

  it('throws on time limit without time plugin', async () => {
    expect(await error('simple', ['--limit', '1 s'])).toMatchSnapshot()
  })

  it('throws on unknown option', async () => {
    expect(await error('unknown')).toMatchSnapshot()
  })

  it('works in integration test with JSON', async () => {
    expect(await check('integration', ['--json'])).toMatchSnapshot()
  })

  it('works in integration test in watch', async () => {
    expect(await check('integration', ['--watch'])).toMatchSnapshot()
  })

  it('shows brotli text when only brotli in config', async () => {
    Object.defineProperty(process, 'version', {
      value: 'v11.7.0'
    })
    expect(await check('brotli')).toMatchSnapshot()
  })

  it('shows brotli text when brotli and gzip in config', async () => {
    Object.defineProperty(process, 'version', {
      value: 'v11.7.0'
    })
    expect(await check('brotli-with-gzip')).toMatchSnapshot()
  })

  it('works in integration test with size', async () => {
    expect(await check('integration')).toMatchSnapshot()
  })

  it('works in integration test with ESM', async () => {
    expect(await check('integration-esm')).toMatchSnapshot()
  })

  it('works in integration test with time', async () => {
    expect(await check('integration', ['--limit', '2s'])).toMatchSnapshot()
  })

  it('shows file not found, package.json', async () => {
    expect(await check('file-not-found')).toMatchSnapshot()
  })

  it('shows file not found, size-limit.json', async () => {
    expect(await check('file-not-found-json-config')).toMatchSnapshot()
  })

  it('shows error when using brotli without webpack', async () => {
    expect(await error('brotli-without-webpack')).toMatchSnapshot()
  })

  it('shows error on time bigger than limit', async () => {
    let [process, history] = createProcess('integration', ['--limit', '1 s'])
    await run(process)
    expect(history.exitCode).toEqual(1)
    expect(history.stderr).toEqual('')
    expect(history.stdout).toMatchSnapshot()
  })

  it('returns zero bytes for empty file', async () => {
    expect(await check('zero')).toMatchSnapshot()
  })

  it('returns zero bytes for empty file without gzip', async () => {
    expect(await check('zero-non-gzip')).toMatchSnapshot()
  })

  it('shows debug', async () => {
    expect(clean(await check('integration', ['--debug']))).toMatchSnapshot()
  })

  it('shows debug on error', async () => {
    expect(clean(await error('internal-error', ['--debug']))).toMatchSnapshot()
  })
})
