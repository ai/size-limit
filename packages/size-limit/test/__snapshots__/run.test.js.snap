// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`run returns zero bytes for empty file 1`] = `
"  
  Size limit: [32m[1m0 B[22m[39m
  Size:       [32m[1m0 B[22m[39m [90mwith all dependencies, minified and gzipped[39m
  
"
`;

exports[`run returns zero bytes for empty file without gzip 1`] = `
"  
  Size limit: [32m[1m0 B[22m[39m
  Size:       [32m[1m0 B[22m[39m [90mwith all dependencies and minified[39m
  
"
`;

exports[`run shows brotli text when brotli and gzip in config 1`] = `
"  
  Size limit:   [32m[1m1 KB[22m[39m
  Size:         [32m[1m-53 B[22m[39m [90mwith all dependencies, minified and brotli[39m
  Loading time: [32m[1m10 ms[22m[39m [90mon slow 3G[39m
  Running time: [32m[1m1 s  [22m[39m [90mon Snapdragon 410[39m
  Total time:   [32m[1m1.1 s[22m[39m
  
"
`;

exports[`run shows brotli text when only brotli in config 1`] = `
"  
  Size limit:   [32m[1m1 KB[22m[39m
  Size:         [32m[1m-53 B[22m[39m [90mwith all dependencies, minified and brotli[39m
  Loading time: [32m[1m10 ms[22m[39m [90mon slow 3G[39m
  Running time: [32m[1m1 s  [22m[39m [90mon Snapdragon 410[39m
  Total time:   [32m[1m1.1 s[22m[39m
  
"
`;

exports[`run shows debug 1`] = `
"{
  \\"cwd\\": \\"/tmp/\\",
  \\"configPath\\": \\"package.json\\",
  \\"checks\\": [
    {
      \\"limit\\": \\"200 B\\",
      \\"files\\": [
        \\"/packages/size-limit/test/fixtures/integration/index.js\\"
      ],
      \\"name\\": \\"index.js\\",
      \\"sizeLimit\\": 200,
      \\"webpackOutput\\": \\"/tmp/\\",
      \\"webpackConfig\\": {
        \\"entry\\": {
          \\"index\\": [
            \\"/packages/size-limit/test/fixtures/integration/index.js\\"
          ]
        },
        \\"output\\": {
          \\"path\\": \\"/tmp/\\"
        },
        \\"optimization\\": {
          \\"concatenateModules\\": true
        },
        \\"resolve\\": {
          \\"plugins\\": [
            {
              \\"topLevelLoader\\": {}
            }
          ]
        },
        \\"resolveLoader\\": {
          \\"plugins\\": [
            {}
          ]
        },
        \\"module\\": {
          \\"rules\\": [
            {
              \\"test\\": {},
              \\"use\\": \\"file-loader\\"
            },
            {
              \\"test\\": {},
              \\"exclude\\": {},
              \\"use\\": [
                \\"style-loader\\",
                \\"css-loader\\"
              ]
            },
            {
              \\"test\\": {},
              \\"use\\": [
                \\"style-loader\\",
                {
                  \\"loader\\": \\"css-loader\\",
                  \\"options\\": {
                    \\"modules\\": true
                  }
                }
              ]
            }
          ]
        },
        \\"plugins\\": [
          {
            \\"pluginDescriptor\\": {
              \\"name\\": \\"OptimizeCssAssetsWebpackPlugin\\"
            },
            \\"options\\": {
              \\"assetProcessors\\": [
                {
                  \\"phase\\": \\"compilation.optimize-chunk-assets\\",
                  \\"regExp\\": {}
                }
              ],
              \\"assetNameRegExp\\": {},
              \\"cssProcessorOptions\\": {},
              \\"cssProcessorPluginOptions\\": {}
            },
            \\"phaseAssetProcessors\\": {
              \\"compilation.optimize-chunk-assets\\": [
                {
                  \\"phase\\": \\"compilation.optimize-chunk-assets\\",
                  \\"regExp\\": {}
                }
              ],
              \\"compilation.optimize-assets\\": [],
              \\"emit\\": []
            },
            \\"deleteAssetsMap\\": {}
          }
        ]
      },
      \\"bundles\\": [
        \\"/tmp/index.js\\"
      ],
      \\"size\\": 31,
      \\"loadTime\\": 0.01,
      \\"runTime\\": 1,
      \\"time\\": 1.01,
      \\"passed\\": true
    }
  ],
  \\"failed\\": false
}
  
  Size limit:   [32m[1m200 B[22m[39m
  Size:         [32m[1m31 B [22m[39m [90mwith all dependencies, minified and gzipped[39m
  Loading time: [32m[1m10 ms[22m[39m [90mon slow 3G[39m
  Running time: [32m[1m1 s  [22m[39m [90mon Snapdragon 410[39m
  Total time:   [32m[1m1.1 s[22m[39m
  
"
`;

exports[`run shows debug on error 1`] = `
"{
  \\"cwd\\": \\"/tmp/\\",
  \\"configPath\\": \\"package.json\\",
  \\"checks\\": [
    {
      \\"limit\\": \\"200 s\\",
      \\"files\\": [
        \\"/packages/size-limit/test/fixtures/internal-error/index.js\\"
      ],
      \\"name\\": \\"index.js\\",
      \\"timeLimit\\": 200
    }
  ]
}
[41m[30m ERROR [39m[49m [31mAdd [33m@size-limit/file[31m plugin to Size Limit[39m
"
`;

exports[`run shows error on missed package.json 1`] = `
"[41m[30m ERROR [39m[49m [31mSize Limit didnâ€™t find [33mpackage.json[31m.
        Create npm package and run Size Limit there.[39m
"
`;

exports[`run shows error on time bigger than limit 1`] = `
"  
  [31mTotal time limit has exceeded[39m
  Time limit:   [31m[1m1 s[22m[39m
  Size:         [31m[1m31 B [22m[39m [90mwith all dependencies, minified and gzipped[39m
  Loading time: [31m[1m10 ms[22m[39m [90mon slow 3G[39m
  Running time: [31m[1m1 s  [22m[39m [90mon Snapdragon 410[39m
  Total time:   [31m[1m1.1 s[22m[39m
  
  [33mTry to reduce size or increase limit in [1m\\"size-limit\\"[22m section of [1mpackage.json[22m[39m
"
`;

exports[`run shows error when using brotli without webpack 1`] = `
"[41m[30m ERROR [39m[49m [31mConfig option [33mbrotli[31m needs [33m@size-limit/webpack[31m plugin[39m
"
`;

exports[`run shows file not found, package.json 1`] = `
"  
  [33mFile not found: folder/index.js[39m
  Size limit: [32m[1m20 KB[22m[39m
  Size:       [32m[1m0 B  [22m[39m [90mgzipped[39m
  
"
`;

exports[`run shows file not found, size-limit.json 1`] = `
"  
  [33mFile not found: folder/index.js[39m
  Size limit: [32m[1m100 KB[22m[39m
  Size:       [32m[1m0 B   [22m[39m [90mgzipped[39m
  
"
`;

exports[`run shows help 1`] = `
"[33msize-limit [OPTION]â€¦ [FILE]â€¦[39m
Check the real performance cost of your front-end project to users

[1mCore options:[22m
  [33m--limit LIMIT[39m  Set size or running time limit for files
  [33m--json[39m         Show results in JSON format
  [33m--help[39m         Display this help
  [33m--watch[39m        Runs in watch mode
  [33m--debug[39m        Show internal configs for issue report
  [33m--version[39m      Display version

[1mExamples:[22m
  [33msize-limit[39m
    Read configuration from [1mpackage.json[22m or [1m.size-limit.json[22m and check limit
[33m  size-limit index.js[39m
    Show the size of specific files
"
`;

exports[`run shows migration guide for npm users 1`] = `
"[31mInstall Size Limit preset depends on type of the project[39m

For application, where you send JS bundle directly to users
  [33mnpm install --save-dev @size-limit/preset-app[39m

For frameworks, components and big libraries
  [33mnpm install --save-dev @size-limit/preset-big-lib[39m

For small (< 10 KB) libraries
  [33mnpm install --save-dev @size-limit/preset-small-lib[39m

Check out docs for more complicated cases
  [33mhttps://github.com/ai/size-limit/[39m
"
`;

exports[`run shows migration guide for npm users: config 1, dep 0 1`] = `
"[31mInstall Size Limit preset depends on type of the project[39m

For application, where you send JS bundle directly to users
  [33mnpm install --save-dev @size-limit/preset-app[39m

For frameworks, components and big libraries
  [33mnpm install --save-dev @size-limit/preset-big-lib[39m

For small (< 10 KB) libraries
  [33mnpm install --save-dev @size-limit/preset-small-lib[39m

Check out docs for more complicated cases
  [33mhttps://github.com/ai/size-limit/[39m

You need to add size-limit dependency: [33mnpm install --save-dev size-limit[39m
"
`;

exports[`run shows migration guide for yarn users 1`] = `
"[31mInstall Size Limit preset depends on type of the project[39m

For application, where you send JS bundle directly to users
  [33myarn add --dev @size-limit/preset-app[39m

For frameworks, components and big libraries
  [33myarn add --dev @size-limit/preset-big-lib[39m

For small (< 10 KB) libraries
  [33myarn add --dev @size-limit/preset-small-lib[39m

Check out docs for more complicated cases
  [33mhttps://github.com/ai/size-limit/[39m
"
`;

exports[`run shows size-limit dependency warning 1`] = `""`;

exports[`run shows webpack-related help 1`] = `
"[33msize-limit [OPTION]â€¦ [FILE]â€¦[39m
Check the real performance cost of your front-end project to users

[1mCore options:[22m
  [33m--limit LIMIT[39m  Set size or running time limit for files
  [33m--json[39m         Show results in JSON format
  [33m--help[39m         Display this help
  [33m--watch[39m        Runs in watch mode
  [33m--debug[39m        Show internal configs for issue report
  [33m--version[39m      Display version

[1mWebpack options:[22m
  [33m--why[39m              Show package content
  [33m--save-bundle DIR[39m  Put build files to check them by hand
  [33m--clean-dir[39m        Remove build files folder before start

[1mExamples:[22m
  [33msize-limit[39m
    Read configuration from [1mpackage.json[22m or [1m.size-limit.json[22m and check limit
[33m  size-limit index.js[39m
    Show the size of specific files with all file dependencies
  [33msize-limit --why[39m
    Show reasons why project have this size
"
`;

exports[`run throws on --clean-dir argument without --save-bundle 1`] = `
"[41m[30m ERROR [39m[49m [31mArgument [33m--clean-dir[31m works only with [33m--save-bundle[31m argument[39m
"
`;

exports[`run throws on --save-bundle argument without DIR parameter 1`] = `
"[41m[30m ERROR [39m[49m [31mMissing parameter [33mDIR[31m for [33m--save-bundle[31m argument[39m
"
`;

exports[`run throws on --save-bundle argument without DIR parameter 2`] = `
"[41m[30m ERROR [39m[49m [31mMissing parameter [33mDIR[31m for [33m--save-bundle[31m argument[39m
"
`;

exports[`run throws on --save-bundle argument without webpack 1`] = `
"[41m[30m ERROR [39m[49m [31mArgument [33m--save-bundle[31m works only with [33m@size-limit/webpack[31m plugin[39m
"
`;

exports[`run throws on --why argument without webpack 1`] = `
"[41m[30m ERROR [39m[49m [31mArgument [33m--why[31m works only with [33m@size-limit/webpack[31m plugin.
        You can add Bundle Analyzer to you own bundler.[39m
"
`;

exports[`run throws on empty config 1`] = `
"[41m[30m ERROR [39m[49m [31mSize Limit config must [33mnot be empty[31m[39m

  [32m\\"size-limit\\"[39m: [
    {
      [32m\\"path\\"[39m: [33m\\"dist/bundle.js\\"[39m,
      [32m\\"limit\\"[39m: [33m\\"10 KB\\"[39m
    }
  ]
"
`;

exports[`run throws on gzip option without gzip plugin 1`] = `
"[41m[30m ERROR [39m[49m [31mConfig option [33mgzip[31m needs [33m@size-limit/webpack[31m or [33m@size-limit/file[31m plugin[39m
"
`;

exports[`run throws on no config 1`] = `
"[41m[30m ERROR [39m[49m [31mCreate Size Limit config in [33mpackage.json[31m[39m

  [32m\\"size-limit\\"[39m: [
    {
      [32m\\"path\\"[39m: [33m\\"dist/bundle.js\\"[39m,
      [32m\\"limit\\"[39m: [33m\\"10 KB\\"[39m
    }
  ]
"
`;

exports[`run throws on non-array config 1`] = `
"[41m[30m ERROR [39m[49m [31mSize Limit config must contain [33man array[31m[39m

  [32m\\"size-limit\\"[39m: [
    {
      [32m\\"path\\"[39m: [33m\\"dist/bundle.js\\"[39m,
      [32m\\"limit\\"[39m: [33m\\"10 KB\\"[39m
    }
  ]
"
`;

exports[`run throws on non-object check 1`] = `
"[41m[30m ERROR [39m[49m [31mSize Limit config array should contain [33monly objects[31m[39m

  [32m\\"size-limit\\"[39m: [
    {
      [32m\\"path\\"[39m: [33m\\"dist/bundle.js\\"[39m,
      [32m\\"limit\\"[39m: [33m\\"10 KB\\"[39m
    }
  ]
"
`;

exports[`run throws on non-string entry 1`] = `
"[41m[30m ERROR [39m[49m [31mThe [33mentry[31m in Size Limit config must be [33ma string[31m or [33man array of strings[31m[39m
"
`;

exports[`run throws on non-string path 1`] = `
"[41m[30m ERROR [39m[49m [31mThe [33mpath[31m in Size Limit config must be [33ma string[31m or [33man array of strings[31m[39m

  [32m\\"size-limit\\"[39m: [
    {
      [32m\\"path\\"[39m: [33m\\"dist/bundle.js\\"[39m,
      [32m\\"limit\\"[39m: [33m\\"10 KB\\"[39m
    }
  ]
"
`;

exports[`run throws on running option without time plugin 1`] = `
"[41m[30m ERROR [39m[49m [31mConfig option [33mrunning[31m needs [33m@size-limit/time[31m plugin[39m
"
`;

exports[`run throws on time limit without time plugin 1`] = `
"[41m[30m ERROR [39m[49m [31mAdd [33m@size-limit/time[31m plugin to use time limit[39m
"
`;

exports[`run throws on unknown argument 1`] = `
"[41m[30m ERROR [39m[49m [31mUnknown argument [33m--unknown[31m.
        Check command for typo and read docs.[39m
"
`;

exports[`run throws on unknown option 1`] = `
"[41m[30m ERROR [39m[49m [31mUnknown option [33munknown[31m in config.
        Check Size Limit docs and version.[39m
"
`;

exports[`run throws on webpack option without webpack plugin 1`] = `
"[41m[30m ERROR [39m[49m [31mConfig option [33mwebpack[31m needs [33m@size-limit/webpack[31m plugin[39m
"
`;

exports[`run throws on webpack option without webpack plugin 2`] = `
"[41m[30m ERROR [39m[49m [31mConfig option [33mignore[31m needs [33m@size-limit/webpack[31m plugin[39m
"
`;

exports[`run throws on webpack option without webpack plugin 3`] = `
"[41m[30m ERROR [39m[49m [31mConfig option [33mconfig[31m needs [33m@size-limit/webpack[31m plugin[39m
"
`;

exports[`run uses dependencies to detect plugins 1`] = `
"[33msize-limit [OPTION]â€¦ [FILE]â€¦[39m
Check the real performance cost of your front-end project to users

[1mCore options:[22m
  [33m--limit LIMIT[39m  Set size or running time limit for files
  [33m--json[39m         Show results in JSON format
  [33m--help[39m         Display this help
  [33m--watch[39m        Runs in watch mode
  [33m--debug[39m        Show internal configs for issue report
  [33m--version[39m      Display version

[1mWebpack options:[22m
  [33m--why[39m              Show package content
  [33m--save-bundle DIR[39m  Put build files to check them by hand
  [33m--clean-dir[39m        Remove build files folder before start

[1mExamples:[22m
  [33msize-limit[39m
    Read configuration from [1mpackage.json[22m or [1m.size-limit.json[22m and check limit
[33m  size-limit index.js[39m
    Show the size of specific files with all file dependencies
  [33msize-limit --why[39m
    Show reasons why project have this size
"
`;

exports[`run works in integration test in watch 1`] = `
"  
  Size limit:   [32m[1m200 B[22m[39m
  Size:         [32m[1m31 B [22m[39m [90mwith all dependencies, minified and gzipped[39m
  Loading time: [32m[1m10 ms[22m[39m [90mon slow 3G[39m
  Running time: [32m[1m1 s  [22m[39m [90mon Snapdragon 410[39m
  Total time:   [32m[1m1.1 s[22m[39m
  
"
`;

exports[`run works in integration test with ESM 1`] = `
"  
  Size limit: [32m[1m1 B[22m[39m
  Size:       [32m[1m1 B[22m[39m [90mwith all dependencies, minified and gzipped[39m
  
"
`;

exports[`run works in integration test with JSON 1`] = `
"[
  {
    \\"name\\": \\"index.js\\",
    \\"passed\\": true,
    \\"size\\": 31,
    \\"running\\": 1,
    \\"loading\\": 0.01
  }
]
"
`;

exports[`run works in integration test with size 1`] = `
"  
  Size limit:   [32m[1m200 B[22m[39m
  Size:         [32m[1m31 B [22m[39m [90mwith all dependencies, minified and gzipped[39m
  Loading time: [32m[1m10 ms[22m[39m [90mon slow 3G[39m
  Running time: [32m[1m1 s  [22m[39m [90mon Snapdragon 410[39m
  Total time:   [32m[1m1.1 s[22m[39m
  
"
`;

exports[`run works in integration test with time 1`] = `
"  
  Time limit:   [32m[1m2 s[22m[39m
  Size:         [32m[1m31 B [22m[39m [90mwith all dependencies, minified and gzipped[39m
  Loading time: [32m[1m10 ms[22m[39m [90mon slow 3G[39m
  Running time: [32m[1m1 s  [22m[39m [90mon Snapdragon 410[39m
  Total time:   [32m[1m1.1 s[22m[39m
  
"
`;
